pkg/services/dashboards/service/integration_test/dashboard_k8s_test.go

```diff
@@ -0,0 +1,1680 @@
+
+		// Try to access the dashboard from a viewer in the other org
+		_, err = otherOrgClient.Resource.Get(context.Background(), dashUID, v1.GetOptions{})
+		require.Error(t, err, "User from other org should not be able to view dashboard even with custom permissions")
+		statusErr := ctx.Helper.AsStatusError(err)
+		require.Equal(t, http.StatusNotFound, int(statusErr.Status().Code), "Should get 404 Not Found")
+
+		// Clean up
+		err = adminClient.Resource.Delete(context.Background(), dashUID, v1.DeleteOptions{})
+		require.NoError(t, err)
+	})
+}
+
+// Run tests for dashboard validations
+func runDashboardValidationTests(t *testing.T, ctx TestContext) {
+	t.Helper()
+
+	adminClient := getResourceClient(t, ctx.Helper, ctx.AdminUser, getDashboardGVR())
+	adminFolderClient := getResourceClient(t, ctx.Helper, ctx.AdminUser, getFolderGVR())
+	editorClient := getResourceClient(t, ctx.Helper, ctx.EditorUser, getDashboardGVR())
+
+	t.Run("Dashboard UID validations", func(t *testing.T) {
+		// Test creating dashboard with existing UID
+		t.Run("reject dashboard with existing UID", func(t *testing.T) {
+			// Create a dashboard with a specific UID
+			specificUID := "existing-uid-dash"
+			createdDash, err := createDashboard(t, adminClient, "Dashboard with Specific UID", nil, &specificUID)
+			require.NoError(t, err)
+
+			// Try to create another dashboard with the same UID
+			_, err = createDashboard(t, adminClient, "Another Dashboard with Same UID", nil, &specificUID)
+			require.Error(t, err)
+
+			// Clean up
+			err = adminClient.Resource.Delete(context.Background(), createdDash.GetName(), v1.DeleteOptions{})
+			require.NoError(t, err)
+		})
+
+		// Test creating dashboard with too long UID
+		t.Run("reject dashboard with too long UID", func(t *testing.T) {
+			// Create a dashboard with a long UID (over 40 chars)
+			longUID := "this-uid-is-way-too-long-for-a-dashboard-uid-12345678901234567890"
+			_, err := createDashboard(t, adminClient, "Dashboard with Long UID", nil, &longUID)
+			require.Error(t, err)
+		})
+
+		// Test creating dashboard with invalid UID characters
+		t.Run("reject dashboard with invalid UID characters", func(t *testing.T) {
+			invalidUID := "invalid/uid/with/slashes"
+			_, err := createDashboard(t, adminClient, "Dashboard with Invalid UID", nil, &invalidUID)
+			require.Error(t, err)
+		})
+	})
+
+	t.Run("Dashboard title validations", func(t *testing.T) {
+		// Test empty title
+		t.Run("reject dashboard with empty title", func(t *testing.T) {
+			_, err := createDashboard(t, adminClient, "", nil, nil)
+			require.Error(t, err)
+		})
+
+		// Test long title
+		t.Run("reject dashboard with excessively long title", func(t *testing.T) {
+			veryLongTitle := "This is an extremely long dashboard title that should exceed the maximum allowed length for dashboard titles in Grafana and therefore should be rejected by the validation system as too long"
```

Comment:
the current limit is 5000 characters 🤯

Reply:
Not sure that will even work in k8s, we can likely limit that more

Reply:
Oh wow. Changed the test